janus.c

```diff
@@ -919,7 +919,7 @@ static int janus_request_check_secret(janus_request *request, guint64 session_id
 			}
 		}
 		/* We consider a request authorized if either the proper API secret or a valid token has been provided */
-		if(!secret_authorized && !token_authorized)
+		if(!(api_secret != NULL && secret_authorized) && !(janus_auth_is_enabled() && token_authorized))
```

Comment:
I'm not sure that's needed. It's easier to change the defaults above, e.g.:

```
gboolean secret_authorized = (api_secret == NULL), token_authorized = !janus_auth_is_enabled();
```

This way the checks are simplified, because what's disabled automatically resolves to `TRUE`. This would also allow us to remove the `Nothing to check` branch entirely and only do something if either mechanism is needed.

Reply:
I think this will not work, because the problem is in the function `is_api_secret_valid` - if it is disabled, it will return TRUE and we are back where we were before.

So the code might look like this:
```c
	gboolean secret_authorized = (api_secret == NULL), token_authorized = !janus_auth_is_enabled();
	if(secret_authorized && token_authorized) {
		/* Nothing to check */
	} else {
		json_t *root = request->message;
		if(!secret_authorized) {
			/* There's an API secret, check that the client provided it */
			json_t *secret = json_object_get(root, "apisecret");
			if(secret && json_is_string(secret) && janus_strcmp_const_time(json_string_value(secret), api_secret)) {
				secret_authorized = TRUE;
			}
		}
		if(!token_authorized) {
			/* The token based authentication mechanism is enabled, check that the client provided it */
			json_t *token = json_object_get(root, "token");
			if(token && json_is_string(token) && janus_auth_check_token(json_string_value(token))) {
				token_authorized = TRUE;
			}
		}
		/* We consider a request authorized if either the proper API secret or a valid token has been provided */
		if(!secret_authorized && !token_authorized)
			return JANUS_ERROR_UNAUTHORIZED;
	}
```

Reply:
Good point, makes sense :+1:

Reply:
Actually, thinking about it my notes were just wrong... the defaults should NOT be `TRUE`, because it means that if only one mechanism is enabled, it will always be bypassed by the `TRUE` on the *other*, disabled, mechanism (e.g., tokens in use but no API secret, wrong token still passes the check thanks to `secret_authorized` being `TRUE`).

I think that was the key reason why we did that change in #2524.

Reply:
You are right, I did not think about that much when implementing the notes :) The previous solution works fine, even though it is a bit more complicated.